addons/im_livechat/static/src/core/web/discuss_client_action_patch.js

```diff
@@ -0,0 +1,22 @@
+/* @odoo-module */
+
+import { DiscussClientAction } from "@mail/core/web/discuss_client_action";
+import { patch } from "@web/core/utils/patch";
+import { useService } from "@web/core/utils/hooks";
+
+patch(DiscussClientAction.prototype, {
+    setup() {
+        super.setup();
+        this.menu = useService("menu");
+    },
+
+    async restoreDiscussThread(props) {
+        await super.restoreDiscussThread(props);
+        const menuId = this.props.action.params?.menu_id;
+        const appData = this.menu.getCurrentApp() || this.menu.getMenu(menuId);
+        const fromLivechatMenu = appData?.xmlid === "im_livechat.menu_livechat_root";
+        if (this.store.discuss.thread?.readonly && !fromLivechatMenu) {
+            this.threadService.setDiscussThread(this.store.discuss.inbox);
+        }
+    },
```

Comment:
I don't understand the point of this override.
1. "not from livechat" should just be the base code
2. there should be no reason to prevent from opening readonly regardless, I don't understand where this is coming from

Reply:
The base code restore the active thread independantly of what triggered Discuss.
Without the override the admin can open a readonly livechat from the session history, leave Livechat, do some stuff and when he opens Discuss from the home screen, the readonly livechat is still there. Juliette mentionned that it feels off from a functional point of view.

Alternatively what I could do to, modify the base implementation of `restoreDiscussThread()` so that it never restore readonly threads and the override would allow it only if the user is coming from Livechat. 
Does that seem better ?